src/util/nodemap/build_nodemap.h

```diff
@@ -497,9 +510,15 @@ static inline int MPIR_NODEMAP_build_nodemap(int sz,
         out_nodemap[i] = j;
     }
 
-odd_even_cliques:
-    if (odd_even_cliques)
-    {
+cliques:
+    if (MPIR_CVAR_NUM_CLIQUES > 1) {
+        for (i = 0; i < sz; ++i) {
+            if (i % MPIR_CVAR_NUM_CLIQUES) {
+                out_nodemap[i] += (g_max_node_id + 1) * (i % MPIR_CVAR_NUM_CLIQUES);
+            }
+        }
+        g_max_node_id = (g_max_node_id + 1) * MPIR_CVAR_NUM_CLIQUES;
```

Comment:
Can you change this line to `g_max_node_id=(g_max_node_id + 1) * MPIR_CVAR_NUM_CLIQUES - 1;`? The "build_nodemap" function gives "maximum node id" in `g_max_node_id`, but after this code `g_max_node_id` will essentially be "number of nodes".

The current MPICH has some inconsistency (in CH3 and CH4) on whether `g_max_node_id` is "maximum node ID" or "number of nodes". Ken and I discussed this and we think we should clarify it as "maximum node ID", and I have a PR pending for it. So, I think it is better to get it right for this patch.